openapi_python_client/templates/property_templates/enum_property.pyi

```diff
@@ -1,27 +1,27 @@
-{% macro construct(property, source) %}
+{% macro construct(property, source, initial_value="None") %}
 {% if property.required %}
 {{ property.python_name }} = {{ property.reference.class_name }}({{ source }})
 {% else %}
-{{ property.python_name }} = None
+{{ property.python_name }} = {{ initial_value }}
 if {{ source }} is not None:
     {{ property.python_name }} = {{ property.reference.class_name }}({{ source }})
 {% endif %}
 {% endmacro %}
 
-{% macro transform(property, source, destination) %}
+{% macro transform(property, source, destination, declare_type=True) %}
 {% if property.required %}
 {% if property.nullable %}
 {{ destination }} = {{ source }}.value if {{ source }} else None
 {% else %}
 {{ destination }} = {{ source }}.value
 {% endif %}
 {% else %}
-{{ destination }}: {{ property.get_type_string() }} = UNSET
+{{ destination }}{% if declare_type %}: {{ property.get_type_string() }}{% endif %} = UNSET
 if not isinstance({{ source }}, Unset):
 {% if property.nullable %}
-    {{ destination }} = {{ source }}.value if {{ source }} else None
+    {{ destination }} = {{ source }} if {{ source }} else None
 {% else %}
-    {{ destination }} = {{ source }}.value
+    {{ destination }} = {{ source }}
 {% endif %}
```

Comment:
I think the `.value`s here are still necessary when converting this to JSON, unless HTTPX automatically converts Enums to their values?

Reply:
This was a typing issue.

Mypy complained when I generated the `model_with_union_property`
```
a_property: Union[Unset, AnEnum, AnIntEnum]
try:
	a_property = UNSET
	if data is not None:
		a_property = AnEnum(data).value
# ...
```
There was a type error, saying I was providing a `str` to a var typed as `Union[Unset, AnEnum, AnIntEnum]`


FWIW, `json.dumps` _does_ convert these enums to their primitive values (both `(str, Enum)` and `IntEnum` types). I'm assuming that's what httpx uses

Reply:
Oh I forgot that we inherit from `(str, Enum)` instead of just `Enum` for this exact reason. So this is great then!